File: lab6/C++/src/AdvancedLogger.cpp
#include "AdvancedLogger.hpp"
#include "iostream"


void AdvancedLogger::Log(const std::string& text, LogLevel level) const {
    std::cout << "[" << GetNowTime("%H:%M:%S") << "]" << "[" << m_levels.at(level) << "] " << text << std::endl;
}
const std::map<LogLevel, std::string> AdvancedLogger::m_levels
{
    {LogLevel::eInfo, "Информация"},
    {LogLevel::eWarn, "Предупреждение"},
    {LogLevel::eDebug, "Отладка"},
    {LogLevel::eError, "Ошибка"}
};

File: lab6/Java/src/logger/AdvancedLogger.java
package logger;
import enums.LogLevel;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Map;

public class AdvancedLogger extends BaseLogger{


    protected Map<LogLevel, String> levels = Map.of(
        LogLevel.INFO, "Информация",
        LogLevel.WARN, "Предупреждение",
        LogLevel.DEBUG, "Отладка",
        LogLevel.ERROR, "Ошибка");

    public void Log(String text, LogLevel level){
        System.out.println(String.format("[%s][%s] %s", GetNowTime(), levels.get(level), text));
    }

    @Override
    public String GetNowTime(){
        return LocalDateTime.now().format(DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm:ss"));
    }
}